[All Hosts] (manifest) Update non-Outlook docs for GA of unified manifest (#4435)
OfficeDev/office-js-docs-pr: docs/develop/localization.md

> # [Unified manifest (developer preview)](#tab/jsonmanifest)

Reviewer: ```suggestion
# [Unified manifest](#tab/jsonmanifest)
```

## Localize extended overrides

> [!NOTE]

> > This section is not applicable if you are using the unified manifest.

Reviewer: ```suggestion
> This section isn't applicable if you're using the unified manifest.
```
